src/cairo_dependencies_graph/import_parser.py

```diff
@@ -1,18 +1,33 @@
-from starkware.cairo.lang.compiler.ast.code_elements import CodeElementFunction, CodeBlock
+from starkware.cairo.lang.compiler.ast.code_elements import CodeElementFunction, CodeBlock, CodeElementImport
 from starkware.cairo.lang.compiler.ast.visitor import Visitor
 
 
 class ImportParser(Visitor):
 
     def __init__(self):
         super().__init__()
+        self.extracted_imports
 
     def _visit_default(self, obj):
         # top-level code is not generated
         return obj
 
-    def visit_CodeElementFunction(self, elm: CodeElementFunction):
-        return super().visit_CodeElementFunction(elm)
-
     def visit_CodeBlock(self, elm: CodeBlock):
+        self.extract_imports(elm)
```

Comment:
Back on this : If we visit all the other codeblocks, that means that we're calling `self.extract_imports(elm)` which overrides the previous value of `self.extracted_imports = res`.
Actually (and sorry for backtracking) I think it's fine if we don't visit the whole file and just return `self.extract_imports(elm)`

Reply:
I think the current solution, where imports are appended to the `self.extracted_imports` is ok